src/action/network.rs

```diff
@@ -0,0 +1,296 @@
+///
+/// [protocol_socket_info]: ../../../netstat2/enum.ProtocolSocketInfo.html
+fn make_connection_from_socket_info<'a>(
+    socket_info: &'a ProtocolSocketInfo
+) -> NetworkConnection {
+    match socket_info {
+        Tcp(tcp_info) => NetworkConnection {
+            family: Some(make_family(&tcp_info.local_addr) as i32),
+            r#type: Some(Type::SockStream as i32),
+            local_address: Some(make_network_endpoint(
+                &tcp_info.local_addr,
+                tcp_info.local_port
+            )),
+            remote_address: Some(make_network_endpoint(
+                &tcp_info.remote_addr,
+                tcp_info.remote_port
+            )),
+            state: Some(make_state(&tcp_info.state) as i32),
+            ..Default::default()
+        },
+        Udp(udp_info) => NetworkConnection {
+            family: Some(make_family(&udp_info.local_addr) as i32),
+            r#type: Some(Type::SockDgram as i32),
+            local_address: Some(make_network_endpoint(
+                &udp_info.local_addr,
+                udp_info.local_port
+            )),
+            remote_address: None,
+            state: None,
+            ..Default::default()
+        }
+    }
+}
+
+impl super::Request for Request {
+
+    type Proto = ListNetworkConnectionsArgs;
+
+    fn from_proto(proto: ListNetworkConnectionsArgs) -> Request {
+        Request {
+            listening_only: proto.listening_only.unwrap_or(false)
+        }
+    }
+}
+
+impl<'a> super::Response for Response<'a> {
+
+    const RDF_NAME: Option<&'static str> = Some("NetworkConnection");
+
+    type Proto = NetworkConnection;
+
+    fn into_proto(self) -> Self::Proto {
+        let mut result: NetworkConnection;
+        result = make_connection_from_socket_info(self.socket_info);
+        if let Some(process_info) = self.process_info {
+            result.pid = Some(process_info.pid);
+            result.process_name = process_info.name;
+        }
+        result
+    }
+}
+
+impl From<&Process> for ProcessInfo {
+
+    fn from(process: &Process) -> ProcessInfo {
+        ProcessInfo {
+            pid: process.pid() as u32,
+            name: Some(process.name().to_string())
+        }
+    }
+}
+
+impl ProcessInfo {
+
+    /// Constucts a ProcessInfo system from a given process ID. The process
+    /// name will not be set.
+    fn from_pid(pid: u32) -> ProcessInfo {
+        ProcessInfo {
+            pid,
+            name: None
+        }
+    }
+
+    /// Constucts a ProcessInfo system from a given process ID. The process
+    /// name will be retrieved from the system using `system` parameter.
+    fn from_system<S: SystemExt>(system: &S, pid: u32) -> ProcessInfo {
+        match system.get_process(pid as i32) {
+            Some(process) => ProcessInfo::from(process),
+            None => ProcessInfo::from_pid(pid)
+        }
+    }
+}
+
+/// Handles requests for the network connection action.
+pub fn handle<S: Session>(
```

Comment:
If you use the `where` syntax for trait bounds, you can make the signature fit in 80 characters (with `where` clause on the next line), like in [`src/action/mod.rs`][1].

[1]: https://github.com/google/rrg/blob/9293f63fff68927e5a045154e40c727236243f66/src/action/mod.rs#L47-L50